koku/masu/database/trino_sql/reporting_ocpazurecostlineitem_daily_summary.sql

```diff
@@ -285,7 +298,322 @@ GROUP BY coalesce(azure.date, azure.usagedatetime),
+        AND ocp.usage_start < date_add('day', 1, {{end_date}})
+        AND azure.ocp_source = {{ocp_source_uuid}}
+        AND azure.year = {{year}}
+        AND azure.month = {{month}}
+)
+SELECT
+    ocp_filtered.azure_partial_resource_id,
+    max(az_disk_capacity.capacity) as capacity,
+    date(coalesce(date, usagedatetime)) as usage_start,
+    {{ocp_source_uuid}} as ocp_source,
+    {{year}} as year,
+    {{month}} as month
+FROM azure_line_items as azure
+JOIN postgres.public.reporting_common_diskcapacity as az_disk_capacity
+    ON azure.metername LIKE '%' || az_disk_capacity.product_substring || ' %' -- space here is important to avoid partial matching
+    AND az_disk_capacity.provider_type = 'Azure'
+JOIN cte_ocp_filtered_resources as ocp_filtered
+    ON split_part(coalesce(nullif(azure.resourceid, ''), azure.instanceid), '/', 9) = ocp_filtered.azure_partial_resource_id
+WHERE coalesce(azure.date, azure.usagedatetime) >= TIMESTAMP '{{start_date | sqlsafe}}'
+    AND coalesce(azure.date, azure.usagedatetime) < date_add('day', 1, TIMESTAMP '{{end_date | sqlsafe}}')
+    AND coalesce(nullif(azure.servicename, ''), azure.metercategory) LIKE '%Storage%'
+    AND coalesce(nullif(azure.resourceid, ''), azure.instanceid) LIKE '%%Microsoft.Compute/disks/%%'
+GROUP BY ocp_filtered.azure_partial_resource_id, date(coalesce(date, usagedatetime))
+{% endif %}
+;
+
+{% if unattributed_storage %}
+-- resource_id matching
+-- Storage disk resources:
+-- (PV’s Capacity) / Disk Capacity * Cost of Disk
+-- PV without PVCs are Unattributed Storage
+INSERT INTO hive.{{schema | sqlsafe}}.reporting_ocpazurecostlineitem_project_daily_summary_temp (
+    azure_uuid,
+    cluster_id,
+    cluster_alias,
+    data_source,
+    namespace,
+    node,
+    persistentvolumeclaim,
+    persistentvolume,
+    storageclass,
+    resource_id,
+    usage_start,
+    usage_end,
+    service_name,
+    instance_type,
+    subscription_guid,
+    subscription_name,
+    resource_location,
+    unit_of_measure,
+    usage_quantity,
+    currency,
+    pretax_cost,
+    markup_cost,
+    pod_cost,
+    project_markup_cost,
+    pod_usage_cpu_core_hours,
+    pod_request_cpu_core_hours,
+    pod_effective_usage_cpu_core_hours,
+    pod_limit_cpu_core_hours,
+    pod_usage_memory_gigabyte_hours,
+    pod_request_memory_gigabyte_hours,
+    pod_effective_usage_memory_gigabyte_hours,
+    node_capacity_cpu_core_hours,
+    node_capacity_memory_gigabyte_hours,
+    cluster_capacity_cpu_core_hours,
+    cluster_capacity_memory_gigabyte_hours,
+    pod_labels,
+    volume_labels,
+    tags,
+    resource_id_matched,
+    cost_category_id,
+    ocp_source,
+    year,
+    month
+)
+SELECT azure.uuid as azure_uuid,
+    max(ocp.cluster_id) as cluster_id,
+    max(ocp.cluster_alias) as cluster_alias,
+    'Storage' as data_source,
+    CASE
+        WHEN max(persistentvolumeclaim) = ''
+            THEN 'Storage unattributed'
+        ELSE max(namespace)
+    END as namespace,
```

Comment:
(question) What is this CASE statement for? When would the `persistentvolumeclaim` be an empty string? Is this when a persistentvolume exists but has no claim?

Reply:
Yeah, this is for when a persistentvolume has no claim it should be counted as a unattributed storage.